investment-app/TRINV.Api/TRINV.Application/Queries/NewsComment/GetAllCommentsByNewsIdQuery.cs

```diff
@@ -0,0 +1,36 @@
+﻿namespace TRINV.Application.Queries.NewsComment;
+
+using Domain.Entities;
+using Interfaces;
+using MediatR;
+using Shared.Business.Exceptions;
+using Shared.Business.Extension;
+using Shared.Business.Utilities;
+
+public record GetAllCommentsByNewsIdQuery(int NewsId) : IRequest<OperationResult<IEnumerable<NewsComment>>>;
+
+internal class GetAllCommentsByNewsIdHandler : IRequestHandler<GetAllCommentsByNewsIdQuery, OperationResult<IEnumerable<NewsComment>>>
+{
+    readonly IRepository<NewsComment> _repository;
+
+    public GetAllCommentsByNewsIdHandler(IRepository<NewsComment> repository)
+    {
+        _repository = repository;
+    }
+
+    public async Task<OperationResult<IEnumerable<NewsComment>>> Handle(GetAllCommentsByNewsIdQuery request, CancellationToken cancellationToken)
+    {
+        var operationResult = new OperationResult<IEnumerable<NewsComment>>();
+
```

Comment:
Shouldn't we validate first if there is news with this id

Reply:
Done.